crates/polars-mem-engine/src/executors/scan/python_scan.rs

```diff
@@ -14,6 +15,39 @@ pub(crate) struct PythonScanExec {
     pub(crate) predicate_serialized: Option<Vec<u8>>,
 }
 
+#[pyclass]
+pub struct PyNodeTimer {
```

Comment:
I don't think it is needed to leak those internals. See comment: crates/polars-mem-engine/src/executors/scan/python_scan.rs